assignment3/bookstore.js

```diff
@@ -0,0 +1,145 @@
+
+const checkedCategory = [];
+const checkboxes = document.querySelectorAll('.check_category');
+const checkAll = document.querySelector('.all');
+const checkWeb = document.querySelector('.web');
+const checkSelf = document.querySelector('.self');
+const checkNovel = document.querySelector('.novel');
+
+let filterBook = [];
+
+//filterBook을 렌더링하는 함수
+const renderBooks = () => {
+    filterBook.forEach((book) => {
+        const bookItem = document.createElement('li');
+        bookItem.classList.add('section_book', 'pulse');
+        bookItem.innerHTML = `
+    <header class="section_book_header"><h2>${book.title}</h2></header>
+    <div class="modal"><p class="modalHash">${book.hashtag}</p> <button class="closeBtn"><img src="images/x-button.png"  width="10" height="10"alt="닫기버튼"></img></button></div>
+    <span class="hashtag_wrapper">
+    <p class="hideHash">${book.hashtag}</p>
+    <button type="button" class="hashPlusButton"><img src="images/plus_button.png" width="25" height="25" alt="플러스버튼"></img></button>
+    </span>
+    <img class="book_img"src="${book.image}" alt="${book.title}이미지">
+    <i class="fas fa-solid fa-heart"></i>
+  `;
+
+        const hashPlusButton = bookItem.querySelector('.hashPlusButton');
+        const modal = bookItem.querySelector('.modal');
+        const closeBtn = bookItem.querySelector('.closeBtn');
+
+        hashPlusButton.addEventListener('click', (e) => {
+            e.preventDefault();
+            modal.style.display = 'flex';
+        });
+
+        // 모달 닫기 버튼에 클릭 이벤트 추가
+        closeBtn.addEventListener('click', () => {
+            // 모달 숨기기
+            modal.style.display = 'none';
+        });
+
+        bookList.appendChild(bookItem);
+    });
+};
+
+//상단 카테고리 리스트를 삭제하는 함수
+const removeCategory = (categoryName, nav_category) => {
+    const p = nav_category.querySelector('p');
+    if (p.textContent === categoryName) {
+        nav_category.remove();
+        const index = checkedCategory.indexOf(categoryName);
+        if (index !== -1) {
+            checkedCategory.splice(index, 1);
+        }
+        filterBooks(checkedCategory);
+    }
+
+    if (categoryName === '전체') {
+        checkAll.checked = false;
+    } else if (categoryName === '웹개발서적') {
+        checkWeb.checked = false;
+    } else if (categoryName === '자기개발서적') {
+        checkSelf.checked = false;
+    } else if (categoryName === '소설') {
+        checkNovel.checked = false;
+    }
+};
+
+//상단에 카테고리를 나열하는 함수
+const renderCategory = (checkedCategory) => {
+    checkedCategory.forEach((categoryName) => {
+        const nav_category = document.createElement('li');
+
+        nav_category.classList.add('nav_category');
+        nav_category.innerHTML = `
+         <p>${categoryName}</p>
+        <button type="button" class="nav_category_button"><img src="./images/x-button.png" alt="닫기버튼" width="10" /></button>
+        `;
+        categoryList.appendChild(nav_category);
```

Comment:
여기두 보안에 취약한 innerHTML 말고 insertAdjacentHTML을 사용한다면 이렇게 리팩토링 할 수 있을 것 같아용!! 

```suggestion
categoryList.insertAdjacentHTML('beforeend', `
  <li class="nav_category">
    <p>${categoryName}</p>
    <button type="button" class="nav_category_button"><img src="./images/x-button.png" alt="닫기버튼" width="10" /></button>
  </li>
`);
```

Reply:
넵 좋은 지적 감사합니다!